### PR TITLE
Add support for Arrow Flight package installation

### PR DESCRIPTION
**Pull request by Rijin-N**
Added an option to install the Arrow Flight package, required for adding Prestissimo support for Arrow Flight connectors (PR: https://github.com/prestodb/presto/pull/24082).

### CODE REVIEW

**Comment by majetideepak** [comment]
@Rijin-N If the connector is being added to Prestissimo, why can't we add these dependencies there?

**Review by assignUser — CHANGES_REQUESTED** [review]
> If the connector is being added to Prestissimo, why can't we add these dependencies there?

This. Please work on decoupling Prestissimos build/dependency system from arrow instead of entangling it more.

File: scripts/setup-ubuntu.sh [hunk 1]
```diff
@@ -130,7 +131,8 @@ function install_velox_deps_from_apt {
     bison \
     flex \
     libfl-dev \
-    tzdata
+    tzdata \
+    pkg-config
```
Comment (czentgr) [inline scripts/setup-ubuntu.sh:135]: This is already added on line 95 and part of the build config. It is not a direct build dependency.

**Comment by Rijin-N** [comment]
> @Rijin-N If the connector is being added to Prestissimo, why can't we add these dependencies there?

@majetideepak Since the Arrow dependencies are already installed in the velox setup scripts and if we want to arrow flight support we need to add it there. If you can suggest a better way to accommodate this in Prestissimo, we can make the changes.

File: scripts/setup-ubuntu.sh [hunk 2]
```diff
@@ -229,6 +231,42 @@ function install_stemmer {
...
+    -DABSL_ENABLE_INSTALL=ON
+}
+
+function install_protobuf {
+  wget_and_untar https://github.com/protocolbuffers/protobuf/releases/download/v21.8/protobuf-all-21.8.tar.gz protobuf
+  (
+    cd ${DEPENDENCY_DIR}/protobuf
+    ./configure --prefix=${INSTALL_PREFIX}
+    make "-j${NPROC}"
+    make install
```
[the remark this answers is not in the document]
Comment (Rijin-N) [reply scripts/setup-ubuntu.sh]: Added ${SUDO}.

File: scripts/setup-ubuntu.sh (hunk 1 above)
Comment (Rijin-N) [reply scripts/setup-ubuntu.sh]: Removed pkg-config. It was not present in some older versions of Velox, which is why I had added it here.

File: scripts/setup-centos9.sh [hunk 3]
```diff
@@ -241,6 +269,8 @@ function install_velox_deps {
   run_and_time install_fbthrift
   run_and_time install_duckdb
   run_and_time install_stemmer
+  run_and_time install_abseil
```
[the remark this answers is not in the document]
Comment (Rijin-N) [reply scripts/setup-centos9.sh:279]: The `install_grpc` function has been moved inside the `install_arrow` function. Other functions were not moved, as they are also required for the GCS connector.

**Comment by majetideepak** [comment]
@Rijin-N Arrow is modular. We can move all of this to the corresponding Prestissimo install scripts and install only arrow flight dependencies (ArrowFlight_LIBRARIES in the Presto PR) there.

**Comment by Rijin-N** [comment]
As suggested by @majetideepak, we have moved the Arrow Flight package installation logic to the Prestissimo install scripts. Hence, we are closing this PR.

**Pull request closed without merging**